irc_client/backend.py

```diff
@@ -272,14 +278,43 @@ def _handle_received_message(self, msg: _ReceivedAndParsedMessage) -> None:
                 self.event_queue.put(SelfChangedNick(old, new))
             else:
                 self.event_queue.put(UserChangedNick(old, new))
+            return
 
-        elif msg.command == "QUIT":
+        if msg.command == "QUIT":
             assert msg.sender is not None
             reason = msg.args[0] if msg.args else None
             self.event_queue.put(UserQuit(msg.sender, reason or None))
-
-        elif msg.sender_is_server:
-            if msg.command == _RPL_NAMREPLY:
+            return
+
+        if msg.sender_is_server:
+            if msg.command == "CAP":
+                subcommand = msg.args[1]
+
+                if subcommand == "ACK":
+                    acknowledged = set(msg.args[-1].split())
+
+                    if "sasl" in acknowledged:
+                        print("SASL was acknowleged.")
+                        self._send_soon("AUTHENTICATE", "PLAIN")
+                elif subcommand == "NAK":
+                    rejected = set(msg.args[-1].split())
+                    if "sasl" in rejected:
+                        print("SASL was rejected.")
+                        raise ValueError("The server does not support SASL.")
```

Comment:
Should this be something better? Can test on ircnet

Reply:
can't get this to happen on ircnet

```suggestion
                        raise ValueError("The server does not support SASL.")  # TODO: this good?
```